File: Services/ClientService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using myClientListApp.Models;
using myClientListApp.Repositories;
using myClientListApp.Services;



public class ClientService : IClientService
{
    private readonly IClientRepository _clientRepository;

    public ClientService(IClientRepository clientRepository)
    {
        _clientRepository = clientRepository;
    }

    public async Task<IEnumerable<Client>> GetAllClients()
    {
        return await _clientRepository.GetAllClients();
    }

    public async Task<Client> GetClientById(int clientId)
    {
        return await _clientRepository.GetClientById(clientId);
    }

    public async Task<IEnumerable<Client>> SearchClients(string query)
    {
        return await _clientRepository.SearchClients(query);
    }


    public async Task AddClient(Client client)
    {
        await _clientRepository.AddClient(client);
    }

    public async Task UpdateClient(Client client)
    {
        await _clientRepository.UpdateClient(client);
    }

    public async Task DeleteClient(int clientId)
    {
        await _clientRepository.DeleteClient(clientId);
    }
}

File: src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { MatDialog } from '@angular/material/dialog';
import { Client } from '../models/client.model';
import { ConfirmationDialogComponent } from '../components/confirmation-dialog/confirmation-dialog.component';


@Injectable({
  providedIn: 'root'
})

export class ConfirmationDialogService {
  constructor(private dialog: MatDialog) { }

  openConfirmationDialog(): Observable<boolean> {
    const dialogRef = this.dialog.open(ConfirmationDialogComponent, {
      width: '250px',
      // Add any other configurations you need
    });

    return dialogRef.afterClosed();
  }
}
export class ClientService {
  private apiUrl = 'https://localhost:7227/api/clients';

  constructor(private http: HttpClient) { }

  getAllClients(): Observable<Client[]> {
    return this.http.get<Client[]>(this.apiUrl);
  }

  getClientById(id: number): Observable<Client> {
    return this.http.get<Client>(`${this.apiUrl}/${id}`);
  }
  searchClients(query: string): Observable<Client[]> {
    const url = `${this.apiUrl}/search?query=${query}`;
    return this.http.get<Client[]>(url);
  }
  addClient(client: Client): Observable<Client> {
    return this.http.post<Client>(this.apiUrl, client);
  }

  updateClient(id: number,client: Client): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}`, client);
  }

  deleteClient(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
